### ISSUE TITLE
"Error 801901F4"/"HTTP 500: Internal Server Error" in Ufasoft miner

### ISSUE THREAD

**Issue by aaferrari** [c00w/bitHopper#330]
A few hours ago I update BitHopper from here, but when I reboot the program, on the <a href=http://bitcointalk.org/index.php?topic=3486.0>Ufasoft miner</a> appears constantly this error (version 0.16): 
<tt>0 MHash /s Error 801901F4</tt>
and with Ufasoft 0.20:
<tt>0 MHash/s     HTTP 500: Internal Server Error</tt>
But before BitHopper update, this miner was working normally and this is the first time that happens to me. I also tried to start the BitHopper with other schedulers and the --p2pLP parameter but Ufasoft miner remains the same.
Finally, from <a href=http://www.fileserve.com/file/53sgwum/bitHopper_debugging.7z>here</a> can download two text files containing the output of the program (with the command line "C:\Python27\python.exe  bitHopper.py --debug 1> debug-bithopper.txt 2> errors-bithooper.txt ", the errors-bithooper.txt was empty, so not included here) if it is necessary to do some error checking or something. One of the files is the complete output of the program (which was running from 23:52:57 until 5:02:28 UTC time) and the other is a more condensed version of the above, but without repeated successive lines).

**Comment by sdogi** [comment]
All I can think is that you added some sort of Bitcoin clones into your list and some of them are sending you crap sometimes. I see this error in logs:

[21:56:42] Error in a wsgi function
[21:56:42] 'NoneType' object has no attribute 'split'

Which is probably causing it.

Could you say which version if bithopper were you running before and what version are you running right now?

From the logs it looks like it works fine for you though.. just that sometimes you get those errors?

**Comment by c00w** [comment]
Our miners assume you supply authentication information. It looks like ufasoft may not be doing that. I'll double check and have it return not authenticated if there is no username or password.

**Comment by aaferrari** [comment]
Well can be say that I can resolve this problem (just the c00w comment gave me the clue to get it).
Sdogi, the error you mention do not know why that happens and I'm not always watching the BitHopper, but I doubt that would cause the problem with the Ufasoft Miner. The BitHopper version I used was between 0.2.4.4 and 0.2.4.5 (I say because the program I had been updated from the repository just before add the bh.cfg file) and now I have version 0.2.4.7. By the way, something I forgot to mention is that in addition to Ufasoft, also I use <a href=http://bitcointalk.org/index.php?topic=1721.0>DiabloMiner</a> (for my graphic card), which works equally good with BitHopper both before and after update (because of which there many "RPC request" in the log).
C00w, if you are referring to the bh.cfg [auth] section, I had understood that this was only to ask for username and password to view statistics in the browser, anyway I activated that option and changed the line command that execute Ufasoft adding the same username and password that the bh.cfg file and although the version 0.16 of Ufasoft not seem to work, trying the same command line in Ufasoft 0.20 curiously worked without problems (being that had previously done the same test with this latest version of the Ufasoft without authentication from the BitHopper but with the same results that in 0.16 version).
To go a close, these are the command lines that normally run with the miners (in case someone else comes to have the same problem):

<ul>
<li>Ufasoft (without authentication): bitcoin-miner.exe -g no -a 1 -v -t 1 -o http://localhost:8337</li>
<li>Ufasoft (with authentication) bitcoin-miner.exe -g no -a 1 -v -t 1 -o http://localhost:8337 -u user -p pass</li>
<li>DiabloMiner (no change in the command line, it works anyway): DiabloMiner-Windows.exe -d -dd  -w 64 -g 1 -D 1 -f 60 -l http://localhost:8337</li>
</ul>

Finally proceed to close this issue because I believe that was solved, but I think it would be good clarify in the README (or elsewhere) that currently the old versions of Ufasoft Miner do not work anymore with BitHopper (if someone, besides me, still retains some old version of that miner, hehe).
And something else, if anyone have any of the errors listed in the logs previously submitted (as the mentioned for sdogi in their respective comment), I suggest open another separate issue ;-).
